### PR TITLE
behavior tests

### PR DESCRIPTION
**Pull request by jhdxr**
TL;DR 
This is a working example of behavior test. Consider workerman as a black box server and verify tests as clients.

As discussed in #904 , I mentioned the behavior test is not doable because pest lacks supporting for isolation process tests. However, after experiments with PhpUnit, I realized [@runinseparateprocess](https://docs.phpunit.de/en/10.0/annotations.html#runinseparateprocess) doesn't help as well. The only programming way to kill the service itself from inside is to kill the whole process, but even for PhpUnit it still requires to do certain jobs after the test method(s). Killing the proces is considered as an error and will leads to a test failure.

This PR proposed a different way for such tests. Instead of trying to asset values on both side, we consider the server side as a black box and only pay attention to the value we sent and received as a client.  To archive this, we create a new process during `beforeAll`/`setUp` (with the help of symfony's `PhpProcess`) and stop it during `afterAll`/`tearDown`. Thus all tests in same file/class can communicate with the server and checking the server's behavior.

---

p.s. a not-so-small suggestion. Tests are not just about adding more test cases. Workerman also needs to be changed/modified to enable better and more comprehensive tests. e.g. When I tried to start workerman, the only way I found is to override `$argv`. There is a `static::$command` and `getArgv` in `Worker` so I guess the author does have an intention to support such use case. However, `parseCommand` still only use `$argv` as the only input, which make `$command` much less useful.

### CODE REVIEW

**Comment by walkor** [comment]
>  Tests are not just about adding more test cases. Workerman also needs to be changed/modified to enable better and more comprehensive tests. e.g. When I tried to start workerman, the only way I found is to override $argv. There is a static::$command and getArgv in Worker so I guess the author does have an intention to support such use case. However, parseCommand still only use $argv as the only input, which make $command much less useful.

That's ok. I [Removed unnecessary $argv](https://github.com/walkor/workerman/commit/36bda0fa84c760c1ed6076a2d473d66222fea697).
Thank you for your post.

**Comment by xpader** [comment]
I have a suggestion before, that is let workerman can run only once like a command, and no master and childprocess, only one process, the process exit after something finished.
Consider to implement that? @walkor

**Comment by jhdxr** [comment]
> I have a suggestion before, that is let workerman can run only once like a command, and no master and childprocess, only one process, the process exit after something finished.

I think what you described is how Workerman works under Windows. 
However, I don't think it's a good idea for testing purpose. We should try to test in an enviroment closing to the actual production.
